### PR TITLE
CLI: CIS docker benchmark 

### PR DESCRIPTION
**Pull request by pbalogh-sa**
Add dockerfile scan to the cli

### CODE REVIEW

File: cli/cmd/scan.go [hunk 1]
```diff
@@ -174,11 +185,16 @@ func vulnerabilityScanner(cmd *cobra.Command, args []string) {
 		logger.Fatalf("Failed get layer commands. %v", err)
 	}
 
+	dockerFileVulnerabilities, err := getDockerFileVulnerabilitiesIfNeeded(sourceType, args[0], appConfig.SharedConfig, dockerFileScan)
```
Comment (FrimIdan) [inline cli/cmd/scan.go:188]: `dockerFileVulnerabilities` --> `cisDockerBenchmarkResults` all places

File: cli/pkg/scanner/export/export.go [hunk 2]
```diff
@@ -55,12 +66,41 @@ func createApplicationVulnerabilityScan(m *scanner.MergedResults, layerCommands
...
 				},
-				ResourceLayerCommands: createResourceLayerCommands(layerCommands),
+				ResourceLayerCommands:     createResourceLayerCommands(layerCommands),
+				CisDockerBenchmarkResults: createCISDockerBenchmarkResults(dockerFileVulnerabilities),
 			},
 		},
 	}
 }
 
+func createCISDockerBenchmarkResults(vulnerabilities dockle_types.AssessmentMap) []*models.CISDockerBenchmarkCodeInfo {
```
Comment (FrimIdan) [inline cli/pkg/scanner/export/export.go:76]: `vulnerabilities` --> `results`

File: cli/cmd/scan.go [hunk 3]
```diff
@@ -72,6 +76,8 @@ func init() {
 		"ID of a defined application to associate the exported vulnerability scan")
 	scanCmd.Flags().BoolP("export", "e", false,
 		"export vulnerability scan results to the backend")
+	scanCmd.Flags().Bool("dockerfile-cis", false,
```
Comment (FrimIdan) [inline cli/cmd/scan.go:79]: maybe `cis-docker-benchmark-scan` , `enables CIS docker benchmark scan. relevant only for image source type`?

File: cli/cmd/scan.go [hunk 4]
```diff
@@ -111,6 +117,11 @@ func vulnerabilityScanner(cmd *cobra.Command, args []string) {
 		logger.Fatalf("Unable to get application ID: %v", err)
 	}
 
+	dockerFileScan, err := cmd.Flags().GetBool("dockerfile-cis")
```
Comment (FrimIdan) [inline cli/cmd/scan.go:120]: `dockerFileScan` --> `cisDockerBenchmarkEnabled`

File: cli/pkg/scanner/export/export.go (hunk 2 above)
Comment (pbalogh-sa) [reply cli/pkg/scanner/export/export.go]: ack

File: cli/cmd/scan.go (hunk 1 above)
Comment (pbalogh-sa) [reply cli/cmd/scan.go]: ack

File: cli/cmd/scan.go (hunk 4 above)
Comment (pbalogh-sa) [reply cli/cmd/scan.go]: ack

File: cli/cmd/scan.go [hunk 5]
```diff
@@ -213,3 +229,39 @@ func getLayerCommandsIfNeeded(sourceType sharedutils.SourceType, source string,
...
+	if sourceType != sharedutils.IMAGE || !needed {
+		return nil, nil
+	}
+
+	assessmentMap, err := dockle_run.RunFromConfig(createDockleConfig(config, source))
+	if err != nil {
+		return nil, fmt.Errorf("failed to run dockle: %w", err)
+	}
+
+	logger.Infof("Image was scanned.")
```
Comment (FrimIdan) [inline cli/cmd/scan.go:246]: Image was scanned for CIS docker benchmark results.

File: cli/cmd/scan.go (hunk 3 above)
Comment (FrimIdan) [reply cli/cmd/scan.go:79]: please also change the flag name `cis-docker-benchmark-scan`

**Comment by FrimIdan** [comment]
LGTM

File: cli/cmd/scan.go [hunk 6]
```diff
@@ -174,11 +185,16 @@ func vulnerabilityScanner(cmd *cobra.Command, args []string) {
 		logger.Fatalf("Failed get layer commands. %v", err)
 	}
 
+	cisDockerBenchmarkResults, err := getCisDockerBenchmarkResultsIfNeeded(sourceType, args[0], appConfig.SharedConfig, cisDockerBenchmarkEnabled)
+	if err != nil {
+		logger.Fatalf("Failed get dockerfile vulnerabilities: %v", err)
```
Comment (akpsgit) [inline cli/cmd/scan.go:190]: Failed to get CIS Docker benchmark results

File: cli/cmd/scan.go [hunk 7]
```diff
@@ -213,3 +229,39 @@ func getLayerCommandsIfNeeded(sourceType sharedutils.SourceType, source string,
...
+
+func createDockleConfig(config *sharedconfig.Config, source string) *dockle_config.Config {
+	var username, password string
+	if len(config.Registry.Auths) > 0 {
+		username = config.Registry.Auths[0].Username
+		password = config.Registry.Auths[0].Password
+	}
+	return &dockle_config.Config{
+		Debug:     log.GetLevel() == log.DebugLevel,
+		Timeout:   60 * time.Second,
```
Comment (akpsgit) [inline cli/cmd/scan.go:260]: I think that we should make it configurable. Also, the default is pretty low, in SCN it's 5 minutes

File: cli/pkg/scanner/export/export.go [hunk 8]
```diff
@@ -55,12 +66,41 @@ func createApplicationVulnerabilityScan(m *scanner.MergedResults, layerCommands
...
+			Assessments: createCISDockerBenchmarkAssesment(info.Assessments),
+			Code:        info.Code,
+			Level:       int64(info.Level),
+		})
+	}
+
+	return ret
+}
+
+func createCISDockerBenchmarkAssesment(assesments dockle_types.AssessmentSlice) []*models.CISDockerBenchmarkAssessment {
```
Comment (akpsgit) [inline cli/pkg/scanner/export/export.go:90]: I know it's a typo also in the API, but need to change assesment-> assessment in all the places in export/export_test :)

Comment (pbalogh-sa) [reply cli/pkg/scanner/export/export.go:90]: thx

File: cli/cmd/scan.go (hunk 7 above)
Comment (pbalogh-sa) [reply cli/cmd/scan.go:260]: ack

**Review by akpsgit — APPROVED** [review]

**Pull request merged**